packages/core/test/interpreter.test.ts

```diff
@@ -1798,7 +1798,7 @@ Event: {\\"type\\":\\"SOME_EVENT\\"}"
       const service = interpret(parentMachine)
         .onTransition(state => {
           if (
-            state.matches('active') &&
+            state.matches<any>('active') &&
```

Comment:
from what I'm observing `parentMachine` is not using type state feature and thus this `state` has its typestate parameter to be `Typestate<any>` and because of that (maybe indirectly?) this `matches` call get inferred to:
```
State<any, AnyEventObject, any, Typestate<any>>.matches<"active">(parentStateValue: "active"): this is never
```
and because of that we can't read `state.children` because `state` got refined by TS to `never`. This seems like regression/bug (not sure sure if introduced here). Unless Typestate is actually provided we probably shouldn't try to utilize it because it leads to situations like this.

Reply:
@davidkpiano this one seems like a problem - if that `<any>` is needed. We don't want people having to do that, right?

Reply:
It doesn't need that, this has been fixed.